crates/ide_assists/src/handlers/generate_function.rs

```diff
@@ -135,6 +136,13 @@ impl FunctionBuilder {
         let fn_name = fn_name(path)?;
         let (type_params, params) = fn_args(ctx, target_module, call)?;
 
+        let await_expr: Option<ast::AwaitExpr> = ctx.find_node_at_offset();
```

Comment:
It would be better to check call expressions parent directly, otherwise we might catch some await expression thats higher up.
Basically `call.syntax().parent().and_then(ast::AwaitExpr::cast)`